### PR TITLE
Update Firefox 124 release notes for WebDriver conforming changes

### PR DESCRIPTION
**Pull request by juliandescottes**
This updates the WebDriver section of the Firefox 124 release notes for all the [note-worthy WebDriver changes in this release](https://bugzilla.mozilla.org/buglist.cgi?v1=fixed&resolution=FIXED&f2=cf_status_firefox123&query_format=advanced&o1=equals&f1=cf_status_firefox124&o2=notequals&status_whiteboard=%5Bwebdriver%3Arelnote%5D&columnlist=component%2Cresolution%2Cassigned_to%2Cshort_desc%2Cbug_type%2Cstatus_whiteboard&status_whiteboard_type=substring&v2=fixed&list_id=16936022).

### CODE REVIEW

**Comment by juliandescottes** [comment]
cc @whimboo

File: files/en-us/mozilla/firefox/releases/124/index.md [hunk 1]
```diff
@@ -60,8 +60,27 @@ This article provides information about the changes in Firefox 124 that affect d
 
 #### WebDriver BiDi
 
+- Implemented the [storage.getCookies](https://w3c.github.io/webdriver-bidi/#command-storage-getCookies) command, which allows users to retrieve cookies. Users can optionally provide `partition` information to retrieve partitioned cookies, as well as `filter` options to only return cookies matching specific criterias. ([Firefox bug 1854580](https://bugzil.la/1854580))
+- Implemented the [storage.setCookie](https://w3c.github.io/webdriver-bidi/#command-storage-setCookie) command, which creates new cookies. Users can provide information about the cookie in the `cookie` parameter, and optionally a `partition` parameter to build the partition key of the partition which should own the cookie. ([Firefox bug 1854582](https://bugzil.la/1854582))
```
Comment (whimboo) [inline files/en-us/mozilla/firefox/releases/124/index.md:64]: nit: This command adds a single cookie.
```suggestion
- Implemented the [storage.setCookie](https://w3c.github.io/webdriver-bidi/#command-storage-setCookie) command, which creates a new cookie. Users can provide information about the cookie in the `cookie` parameter, and optionally a `partition` parameter to build the partition key of the partition which should own the cookie. ([Firefox bug 1854582](https://bugzil.la/1854582))
```

File: files/en-us/mozilla/firefox/releases/124/index.md [hunk 2]
```diff
@@ -60,8 +60,27 @@ This article provides information about the changes in Firefox 124 that affect d
...
+  - [network.failRequest](https://w3c.github.io/webdriver-bidi/#command-network-failRequest) allows to cancel requests intercepted in the `BeforeRequestSent` or the `ResponseStarted` phases.([Firefox bug 1853883](https://bugzil.la/1853883))
+  - [network.continueRequest](https://w3c.github.io/webdriver-bidi/#command-network-continueRequest), [network.continueResponse](https://w3c.github.io/webdriver-bidi/#command-network-continueResponse) and [network.provideResponse](https://w3c.github.io/webdriver-bidi/#command-network-provideResponse) are also available, but at the moment they only support the "request" argument and will only allow to resume a blocked request. In future releases, additional arguments will allow to modify the request and response. ([Firefox bug 1874206](https://bugzil.la/1874206) and [Firefox bug 1853882](https://bugzil.la/1853882))
+- Implemented several commands around "user contexts", implemented as containers in Firefox:
```
Comment (whimboo) [inline files/en-us/mozilla/firefox/releases/124/index.md:71]: Maybe we can link to https://support.mozilla.org/en-US/kb/how-use-firefox-containers?

File: files/en-us/mozilla/firefox/releases/124/index.md [hunk 3]
```diff
@@ -60,8 +60,27 @@ This article provides information about the changes in Firefox 124 that affect d
...
+  - [browser.getUserContexts](https://w3c.github.io/webdriver-bidi/#command-browser-getUserContexts) allows clients to retrieve the list of all available user contexts, including the default user context. ([Firefox bug 1870847](https://bugzil.la/1870847))
+- Added support for the "userContext" argument to the [browsingContext.create](https://w3c.github.io/webdriver-bidi/#command-browsingContext-create) command, which allows to assign a new browsing context (tab or window) to a specific user context (Firefox container). ([Firefox bug 1874918](https://bugzil.la/1874918))
+- Updated the [browsingContext.Info](https://w3c.github.io/webdriver-bidi/#type-browsingContext-Info) type to include the "userContext" field, which is the unique id of the user context owning the tab for this browsing context. ([Firefox bug 1874920](https://bugzil.la/1874920))
+- Added support for the "contexts" argument to the [script.addPreloadScript](https://w3c.github.io/webdriver-bidi/#command-script-addPreloadScript) command, this allows clients to add preload scripts only to specific browsing contexts. ([Firefox bug 1858458](https://bugzil.la/1858458))
```
Comment (whimboo) [inline files/en-us/mozilla/firefox/releases/124/index.md:77]: I think that we should mention top-level browsing context here.
 Maybe something like the following?

```suggestion
- Added support for the "contexts" argument to the [script.addPreloadScript](https://w3c.github.io/webdriver-bidi/#command-script-addPreloadScript) command, this allows clients to add preload scripts only to specific browsing context trees (tabs) by specifying their top-level browsing context. ([Firefox bug 1858458](https://bugzil.la/1858458))
```

File: files/en-us/mozilla/firefox/releases/124/index.md [hunk 4]
```diff
@@ -60,8 +60,27 @@ This article provides information about the changes in Firefox 124 that affect d
...
+- Fixed a bug where [browsingContext.close](https://w3c.github.io/webdriver-bidi/#command-browsingContext-close) would not be able to close the last tab of a window. ([Firefox bug 1873948](https://bugzil.la/1873948))
+
 #### Marionette
 
+- Fixed an issue with [Get Element Text](https://www.w3.org/TR/webdriver2/#get-element-text), which ignored the slot value for web component without custom text. ([Firefox bug 1865381](https://bugzil.la/1865381))
```
Comment (whimboo) [inline files/en-us/mozilla/firefox/releases/124/index.md:82]:
```suggestion
- Fixed an issue with [Get Element Text](https://w3c.github.io/webdriver/#dfn-get-element-text), which ignored the slot value of a web component when no custom text is specified. ([Firefox bug 1865381](https://bugzil.la/1865381))
```

**Review by whimboo — CHANGES_REQUESTED** [review]
Wow, that's a lot of changes for this release! Thanks for writing this all up. It looks all fine and I have only some nits. Please have a look.

File: files/en-us/mozilla/firefox/releases/124/index.md (hunk 1 above)
Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/124/index.md]: Done

File: files/en-us/mozilla/firefox/releases/124/index.md (hunk 2 above)
Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/124/index.md]: Done

File: files/en-us/mozilla/firefox/releases/124/index.md (hunk 3 above)
Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/124/index.md]: Done

File: files/en-us/mozilla/firefox/releases/124/index.md (hunk 4 above)
Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/124/index.md]: Done

**Comment by juliandescottes** [comment]
@whimboo thanks for the review! all comments should be addressed now. Back to you for review.

**Review by whimboo — APPROVED** [review]
Looks fine to me. @lutien do you want to check as well?

File: files/en-us/mozilla/firefox/releases/124/index.md [hunk 5]
```diff
@@ -60,8 +60,27 @@ This article provides information about the changes in Firefox 124 that affect d
 
 #### WebDriver BiDi
 
+- Implemented the [storage.getCookies](https://w3c.github.io/webdriver-bidi/#command-storage-getCookies) command, which allows users to retrieve cookies. Users can optionally provide `partition` information to retrieve partitioned cookies, as well as `filter` options to only return cookies matching specific criterias. ([Firefox bug 1854580](https://bugzil.la/1854580))
```
Comment (lutien) [inline files/en-us/mozilla/firefox/releases/124/index.md:63]: I think saying here `partitioned cookies` might be a bit misleading, so maybe we say something "cookies from a certain partition" (or something similar to what we say for `storage.setCookie`)?

Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/124/index.md:63]: Do users actually know what a partition is? Maybe we should explain that as well?

Comment (lutien) [reply files/en-us/mozilla/firefox/releases/124/index.md:63]: I guess we could maybe link to mdn article about [State_Partitioning](https://developer.mozilla.org/en-US/docs/Web/Privacy/State_Partitioning)?

Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/124/index.md]: I tried to rephrase this section and added the suggested link. Does it look ok now?

Comment (lutien) [reply files/en-us/mozilla/firefox/releases/124/index.md]: Yes, thanks, looks great now!

**Review by lutien — APPROVED** [review]

**Pull request merged**